L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCConcentratorData.cc

```diff
@@ -0,0 +1,155 @@
+                  "HGCalTriggerNtupleHGCConcentratorData");
+
+HGCalTriggerNtupleHGCConcentratorData::HGCalTriggerNtupleHGCConcentratorData(const edm::ParameterSet& conf)
+    : HGCalTriggerNtupleBase(conf) {}
+
+void HGCalTriggerNtupleHGCConcentratorData::initialize(TTree& tree,
+                                                       const edm::ParameterSet& conf,
+                                                       edm::ConsumesCollector&& collector) {
+  concentrator_data_token_ =
+      collector.consumes<l1t::HGCalConcentratorDataBxCollection>(conf.getParameter<edm::InputTag>("ConcentratorData"));
+
+  std::string prefix(conf.getUntrackedParameter<std::string>("Prefix", "econ"));
+
+  std::string bname;
+  auto withPrefix([&prefix, &bname](char const* vname) -> char const* {
+    bname = prefix + "_" + vname;
+    return bname.c_str();
+  });
+
+  tree.Branch(withPrefix("n"), &econ_n_, (prefix + "_n/I").c_str());
+  tree.Branch(withPrefix("id"), &econ_id_);
+  tree.Branch(withPrefix("subdet"), &econ_subdet_);
+  tree.Branch(withPrefix("zside"), &econ_side_);
+  tree.Branch(withPrefix("layer"), &econ_layer_);
+  tree.Branch(withPrefix("wafer"), &econ_wafer_);
+  tree.Branch(withPrefix("waferu"), &econ_waferu_);
+  tree.Branch(withPrefix("waferv"), &econ_waferv_);
+  tree.Branch(withPrefix("wafertype"), &econ_wafertype_);
+  tree.Branch(withPrefix("panel_number"), &econ_panel_number_);
+  tree.Branch(withPrefix("panel_sector"), &econ_panel_sector_);
+  tree.Branch(withPrefix("index"), &econ_index_);
+  tree.Branch(withPrefix("data"), &econ_data_);
+}
+
+void HGCalTriggerNtupleHGCConcentratorData::fill(const edm::Event& e, const edm::EventSetup& es) {
+  // retrieve trigger cells
+  edm::Handle<l1t::HGCalConcentratorDataBxCollection> concentrator_data_h;
+  e.getByToken(concentrator_data_token_, concentrator_data_h);
+  const l1t::HGCalConcentratorDataBxCollection& concentrator_data = *concentrator_data_h;
+
+  // retrieve geometry
+  es.get<CaloGeometryRecord>().get(geometry_);
+
+  triggerTools_.eventSetup(es);
+
+  clear();
+  for (auto econ_itr = concentrator_data.begin(0); econ_itr != concentrator_data.end(0); econ_itr++) {
+    econ_n_++;
+    // hardware data
+    DetId panelId(econ_itr->detId());
+    int panel_sector = -999;
+    int panel_number = -999;
+    if (panelId.det() == DetId::Forward) {
+      HGCalDetId panelIdHGCal(panelId);
+      if (panelId.subdetId() == ForwardSubdetector::HGCHEB) {
+        panel_number = panelIdHGCal.wafer();
+      } else {
+        panel_sector = (panelIdHGCal.wafer() >> kSectorOffset_) & kSectorMask_;
+        panel_number = (panelIdHGCal.wafer() >> kPanelOffset_) & kPanelMask_;
+      }
+    } else if (panelId.det() == DetId::HGCalHSc) {
+      HGCScintillatorDetId panelIdSci(panelId);
+      panel_sector = panelIdSci.iphi();
+      panel_number = panelIdSci.ietaAbs();
+    }
+    econ_panel_number_.emplace_back(panel_number);
+    econ_panel_sector_.emplace_back(panel_sector);
```

Comment:
Given the defintion of the `HGCalConcentratorData` id, this should always be undefined (`-999`), so the panel sector and number can probably be removed from the ntuples